hotel-booking-agent-semantic-kernel/README.md

```diff
@@ -0,0 +1,288 @@
+# Building Secured AI Agents
+
+This sample demonstrates how to build AI assistants that can securely access protected APIs using OAuth authentication.
+The example shows a hotel booking assistant that interacts with secured APIs.
+
+## Overview
+
+The project consists of:
+
+1. **Authentication SDK** (`auth.py`) - Handles OAuth flows including client credentials and user authorization
+2. **Service Layer** (`service.py`) - Implements a WebSocket API for the hotel booking assistant
+3. **API Tools** (`tools.py`) - API client functions for hotel services
+
+## Prerequisites
+
+- Python 3.10+
+- Azure OpenAI service access
+- Asgardeo tenant or other OAuth 2.0 provider
+- (Hotel API service)[https://github.com/nadheesh/agent_demo_hotel_api]
+
+## Configuration Setup
+
+### Asgardeo Configuration (OAuth Provider)
+
+1. **Create an Asgardeo Account**:
+    - Sign up at [https://wso2.com/asgardeo/](https://wso2.com/asgardeo/)
+    - Create a new organization/tenant
+
+2. **Create an Application**:
+    - Navigate to Applications > New Application
+    - Select "Standard-Based Application (OAuth/OIDC)"
+
+3. **Configure the Application**:
+    - **Name**: Hotel Booking Assistant
+    - **Grant Types**: Check "Code", "Client Credentials" and "Refresh"
+    - **Callback URLs**: Add `http://localhost:8000/callback` (or your deployment URL)
+    - **Access Token Type**: JWT
+    - **Allowed Origins**: Add your frontend domain for CORS
+
+4. **Configure Scopes**:
+    - Navigate to "Scopes" section
+    - Create custom scopes for the Hotel API:
+        - `read_hotels`: Permission to view hotel listings
+        - `read_rooms`: Permission to view room details
+        - `create_bookings`: Permission to make reservations
+    - Assign these scopes to your application
+
+5. **Get Credentials**:
+    - After creating the app, copy the "Client ID" and "Client Secret"
+    - Use these in your environment variables (`ASGARDEO_CLIENT_ID` and `ASGARDEO_CLIENT_SECRET`)
+
+### Azure OpenAI Configuration
+
+1. **Create an Azure OpenAI Resource**:
```

Comment:
Can support the example to use Google AI Key, it's free for everyone and convenient for anyone to try out the sample.